controllers/controller/workspacerouting/workspacerouting_controller.go

```diff
@@ -114,7 +114,11 @@ func (r *WorkspaceRoutingReconciler) Reconcile(req ctrl.Request) (ctrl.Result, e
 	}
 
 	restrictedAccess, setRestrictedAccess := instance.Annotations[config.WorkspaceRestrictedAccessAnnotation]
-	routingObjects := solver.GetSpecObjects(instance, workspaceMeta)
+	routingObjects, err := solver.GetSpecObjects(instance, workspaceMeta)
+	if err != nil {
+		return reconcile.Result{}, err
```

Comment:
One thing to think about here is whether we should support failing the workspace start from this error (either via an error type or via another return value). This can come as a separate PR, though.

Reply:
The general flows we should think about are
1. Unexpected error -> return error
2. Expected, transient error (something's not ready) -> print message, return `reconcile.Result{RequeueAfter: 1 * time.Second}, nil`
3. Expected, fatal error (something's not configured) -> fail workspace startup; print message.

Reply:
I like the idea very much. How does one fail workspace startup though?

Reply:
I've added new changes that implement at least the "controller not ready, please retry" part..

Reply:
The additions look great so far! If you want to fail workspace startup, you can do what we do if e.g. `RoutingNotSupported` is returned above:

```
		instance.Status.Phase = controllerv1alpha1.RoutingFailed
		statusErr := r.Status().Update(ctx, instance)
		return reconcile.Result{}, statusErr
```
This will be read by the DevWorkspace controller on the next reconcile, which will cause the overall DW to fail to start. We also have a TODO to propagate some sort of status message from the routing to specify what the issue is; I'll get to that sometime soon hopefully.